python/dataset_plotting.py

```diff
@@ -0,0 +1,239 @@
+    try:
+        from ipywidgets import interactive, HBox, VBox
+    except ImportError:
+        print("Sorry, the sliceviewer requires ipywidgets which was not found on this system")
+        return
+
+    dims = input_data.dimensions()
+    if (len(dims) > 2) and (len(dims) < 5):
+
+        # Get spatial extents by iterating over the coordinates, and also search
+        # for data fields
+        axmin = []
+        axmax = []
+        axlab = []
+        fields = []
+        nx = []
+        ifield = 0
+        for var in input_data:
+            if var.is_coord:
+                values = var.numpy
+                axmin.append(values[0] - 0.5*(values[1]-values[0]))
+                axmax.append(values[-1] + 0.5*(values[-1]-values[-2]))
+                axlab.append([var.name, var.unit])
+            elif var.is_data:
+                fields.append([var.name, var.unit])
+                if var.name == field:
+                    ifield = len(fields) - 1
+
+        ratio = (axmax[1] - axmin[1]) / (axmax[0] - axmin[0])
+
+        if (field is None) and (len(fields) > 1):
+            raise RuntimeError("More than one data field found! Please specify which one to display")
+
+        a = input_data[Data.Value, fields[ifield][0]].numpy
+        nx = np.shape(a)
+
+        fig = go.FigureWidget(
+
+            data = [go.Heatmap(
+                z = a[:,:,0],
+                colorscale = 'Viridis',
+                colorbar=dict(
+                    title="{} [{}]".format(fields[ifield][0],fields[ifield][1]),
+                    titleside = 'right',
+                    )
+                )],
+
+            layout = dict(
+                autosize=False,
+                width=800,
+                height=800*ratio,
+                xaxis = dict(
+                    range = [axmin[0],axmax[0]],
+                    title = "{} [{}]".format(axlab[0][0],axlab[0][1])),
+                yaxis = dict(
+                    scaleanchor = "x",
+                    scaleratio = ratio,
+                    range = [axmin[1],axmax[1]],
+                    title = "{} [{}]".format(axlab[1][0],axlab[1][1]))
+            )
+
+        )
+
+        def update_z(zpos):
+            fig.data[0].z = a[:,:,zpos]
+
+        # Add a slider that updates the slice plane
+        slider = interactive(update_z, zpos=(0, nx[2]-1, 1))
+        vb = VBox((fig, slider))
+        vb.layout.align_items = 'center'
+        return vb
+    else:
+        raise RuntimeError("Unsupported number of dimensions in sliceviewer.")
+
+#===============================================================================
+
+# Plot a 1D spectrum
+# Inputs can be:
+# 1. d (=dataset)
+# 2. d, var (=string)
+# 3. [d1, d2]
+# 4. [[d1, var1], [d2, var2]]
+# 5. [[d1, var1], [d2]]
+#
+# TODO: find a more general way of handling arguments to be sent to plotly,
+# probably via a dictionay of arguments
+def plot_1d(input_data, var=None, logx=False, logy=False, logxy=False):
```

Comment:
I would avoid having a separate `var` argument, just use the slicing/subsetting built-in we have, e.g., pass `d1[tag1, name1]`. That should significantly simply the code below.

Maybe we can also avoid some initial headaches by just supporting a single `DatasetSlice` (with potentially multiple variables). Otherwise you'll have to write code for checking compatible axes, ...
If people really want to plot data from two datasets, maybe they can just merge into one dataset. The data should be small anyway in this case here.

Reply:
Edit: This should read `d1.subset(tag1, name1)`, otherwise you lose the coordinates.

Reply:
Yes, this was all done so we could have some flexible handling of multiple datasets, but i agree that just having subsets in the argument would be better.
So the valid inputs would be:
```Python
plot_1d(d1)
plot_1d([d1, d2])
plot_1d(d1.subset(Data.Value, "Counts"))
plot_1d([d1.subset(Data.Value, "Counts1"), d2.subset(Data.Value, "Counts2")])
```

Reply:
Yes, but I'd ignore case 2 and 4 for simplicity at this point.